File: StudentGradingSystem.Api/Models/Student.cs
namespace StudentGradingSystem.Api.Models;

public class Student
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public int Age { get; set; }

    public int DepartmentId { get; set; }

    public Department Department { get; set; } = null!;

    public double CGPA { get; set; }

    public bool IsDeleted { get; set; } = false;

    public DateTime? DeletedAt { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public string? CreatedBy { get; set; }

    public string? UpdatedBy { get; set; }
}

File: student-grading-system-web/src/types/student.ts
export interface Student {
  id: number;
  name: string;
  age: number;
  rollNumber: string;
  email: string;
  cgpa: number;
  departmentId: number;
  departmentName: string;
}

export interface CreateStudentRequest {
  name: string;
  age: number;
  rollNumber: string;
  email: string;
  cgpa: number;
  departmentId: number;
}

export interface UpdateStudentRequest {
  name: string;
  age: number;
  rollNumber: string;
  email: string;
  cgpa: number;
  departmentId: number;
}